### PR TITLE
Fix(Auth): Add missing instructions in java for SignInNextSteps

### PR DESCRIPTION
**Pull request by gpanshu**
_Issue #, if available:_

_Description of changes:_
Add missing instructions in java for SignInNextSteps

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx [hunk 1]
```diff
@@ -4,9 +4,78 @@ The `nextStep` property is of enum type `AuthSignInStep`. Depending on its value
 
 <BlockSwitcher>
 
+<Block name="Java">
+
+```java
+try {
+            AWSCognitoAuthSignInOptions options =
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx:11]: Please fix the indentation of the code inside this try block.

File: src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx [hunk 2]
```diff
@@ -4,9 +4,78 @@ The `nextStep` property is of enum type `AuthSignInStep`. Depending on its value
...
+            Amplify.Auth.signIn(
+                    "username",
+                    "password",
+                    options,
+                    result ->
+                    {
+                        AuthNextSignInStep nextStep = result.getNextStep();
+                        switch (nextStep.getSignInStep()) {
+                            case CONFIRM_SIGN_IN_WITH_SMS_MFA_CODE: {
+                                Log.i(TAG, "SMS code sent to "+nextStep.getCodeDeliveryDetails().getDestination());
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx]:
```suggestion
                                Log.i(TAG, "SMS code sent to " + nextStep.getCodeDeliveryDetails().getDestination());
```

File: src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx [hunk 3]
```diff
@@ -4,9 +4,78 @@ The `nextStep` property is of enum type `AuthSignInStep`. Depending on its value
...
+                                break;
+                            }
+                            case DONE: {
+                                Log.i(TAG, "SignIn complete");
+                                // User has successfully signed in to the app
+                                break;
+                            }
+                        }
+                    }
+                    ,
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx:65]: This comma can be moved up to line 64 after the `}`

File: src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx [hunk 4]
```diff
@@ -4,9 +4,78 @@ The `nextStep` property is of enum type `AuthSignInStep`. Depending on its value
...
+                    error -> Log.e(TAG, "SignIn failed: "+error.getStackTrace())
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx]:
```suggestion
                    error -> Log.e(TAG, "SignIn failed: " + error.getStackTrace())
```

File: src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx [hunk 5]
```diff
@@ -4,9 +4,78 @@ The `nextStep` property is of enum type `AuthSignInStep`. Depending on its value
...
+            );
+        } catch (Exception error) {
+            Log.e(TAG, "Unexpected error occurred: "+error.getStackTrace());
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx]:
```suggestion
            Log.e(TAG, "Unexpected error occurred: " + error.getStackTrace());
```

File: src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx [hunk 6]
```diff
@@ -4,6 +4,30 @@ Note: the signIn result also includes an `AuthCodeDeliveryDetails` member. It in
 
 <BlockSwitcher>
 
+<Block name="Java">
+
+```java
+try{
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx:10]:
```suggestion
try {
```

File: src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx [hunk 7]
```diff
@@ -4,6 +4,30 @@ Note: the signIn result also includes an `AuthCodeDeliveryDetails` member. It in
...
+                if (result.isSignedIn()) {
+                    Log.i(TAG, "Confirm signIn succeeded");
+                } else {
+                    Log.i(TAG, "Confirm sign in not complete. There might be additional steps: "+ result.getNextStep());
+                    // Switch on the next step to take appropriate actions.
+                    // If `signInResult.isSignedIn` is true, the next step
+                    // is 'done', and the user is now signed in.
+                }
+            }
+  , error -> Log.e(TAG, "Confirm sign in failed: "+error)
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx:23]: Please move the `,` up to line 22

File: src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx [hunk 8]
```diff
@@ -2,6 +2,30 @@ If the next step is `CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE`, Amplify Auth is awa
 
 <BlockSwitcher>
 
+<Block name="Java">
+
+```java
+try{
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx:8]:
```suggestion
try {
```

File: src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx [hunk 9]
```diff
@@ -2,6 +2,30 @@ If the next step is `CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE`, Amplify Auth is awa
...
+                if (result.isSignedIn()) {
+                    Log.i(TAG, "Confirm signIn succeeded");
+                } else {
+                    Log.i(TAG, "Confirm sign in not complete. There might be additional steps: "+ result.getNextStep());
+                    // Switch on the next step to take appropriate actions.
+                    // If `signInResult.isSignedIn` is true, the next step
+                    // is 'done', and the user is now signed in.
+                }
+            }
+  , error -> Log.e(TAG, "Confirm sign in failed: "+error)
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx:21]: Please move the `,` up to line 20.

File: src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx [hunk 10]
```diff
@@ -2,6 +2,30 @@ If the next step is `CONFIRM_SIGN_IN_WITH_NEW_PASSWORD`, Amplify Auth requires a
 
 <BlockSwitcher>
 
+<Block name="Java">
+
+```java
+try{
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx:8]:
```suggestion
try {
```

File: src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx [hunk 11]
```diff
@@ -2,6 +2,30 @@ If the next step is `CONFIRM_SIGN_IN_WITH_NEW_PASSWORD`, Amplify Auth requires a
...
+                if (result.isSignedIn()) {
+                    Log.i(TAG, "Confirm signIn succeeded");
+                } else {
+                    Log.i(TAG, "Confirm sign in not complete. There might be additional steps: "+ result.getNextStep());
+                    // Switch on the next step to take appropriate actions.
+                    // If `signInResult.isSignedIn` is true, the next step
+                    // is 'done', and the user is now signed in.
+                }
+            }
+  , error -> Log.e(TAG, "Confirm sign in failed: "+ error)
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx:21]: Please move the `,` to line 20

File: src/fragments/lib/auth/android/signin_next_steps/50_reset_password.mdx [hunk 12]
```diff
@@ -1,6 +1,21 @@
...
 <BlockSwitcher>
 
+<Block name="Java">
+
+```java
+try {
+    Amplify.Auth.resetPassword(
+            "username",
+            result -> Log.i(TAG, "Reset password succeeded"),
+            error -> Log.e(TAG, "Reset password failed : " + error
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/50_reset_password.mdx:11]:
```suggestion
            error -> Log.e(TAG, "Reset password failed : " + error)
```

File: src/fragments/lib/auth/android/signin_next_steps/50_reset_password.mdx [hunk 13]
```diff
@@ -1,6 +1,21 @@
...
+    );
+    } catch (Exception error) {
+        Log.e(TAG, "Unexpected error: " + error);
+    }
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/50_reset_password.mdx:15]: This catch block is indented too far (it should line up with `try {`)

**Review by eeatonaws — CHANGES_REQUESTED** [review]
Left some comments about formatting and also an error in one of the code snippets. Please also add a space before and after the `+` in the log statements.

File: src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx (hunk 1 above)
Comment (gpanshu) [reply src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx]: done

File: src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx (hunk 3 above)
Comment (gpanshu) [reply src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx]: done

File: src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx (hunk 9 above)
Comment (gpanshu) [reply src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx]: done

File: src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx (hunk 11 above)
Comment (gpanshu) [reply src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx]: done

File: src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx (hunk 10 above)
Comment (gpanshu) [reply src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx:8]: done

File: src/fragments/lib/auth/android/signin_next_steps/50_reset_password.mdx (hunk 12 above)
Comment (gpanshu) [reply src/fragments/lib/auth/android/signin_next_steps/50_reset_password.mdx]: done

File: src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx [hunk 14]
```diff
@@ -4,6 +4,30 @@ Note: the signIn result also includes an `AuthCodeDeliveryDetails` member. It in
...
+
+```java
+try {
+    Amplify.Auth.confirmSignIn(
+            "confirmation code",
+            result -> {
+                if (result.isSignedIn()) {
+                    Log.i(TAG, "Confirm signIn succeeded");
+                } else {
+                    Log.i(TAG, "Confirm sign in not complete. There might be additional steps: "+ result.getNextStep());
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx:17]:
```suggestion
                    Log.i(TAG, "Confirm sign in not complete. There might be additional steps: " + result.getNextStep());
```

File: src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx [hunk 15]
```diff
@@ -2,6 +2,30 @@ If the next step is `CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE`, Amplify Auth is awa
...
+                    Log.i(TAG, "Confirm signIn succeeded");
+                } else {
+                    Log.i(TAG, "Confirm sign in not complete. There might be additional steps: " + result.getNextStep());
+                    // Switch on the next step to take appropriate actions.
+                    // If `signInResult.isSignedIn` is true, the next step
+                    // is 'done', and the user is now signed in.
+                }
+            },
+             error -> Log.e(TAG, "Confirm sign in failed: " + error)
+            );
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx:22]:
```suggestion
      );
```

File: src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx [hunk 16]
```diff
@@ -4,6 +4,30 @@ Note: the signIn result also includes an `AuthCodeDeliveryDetails` member. It in
...
+                    // Switch on the next step to take appropriate actions.
+                    // If `signInResult.isSignedIn` is true, the next step
+                    // is 'done', and the user is now signed in.
+                }
+            },
+             error -> Log.e(TAG, "Confirm sign in failed: " + error)
+            );
+    } catch (Exception error) {
+        Log.e(TAG, "Unexpected error: "+error);
+    }
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx:27]: This catch block is indented too far (it should line up with `try {`)

File: src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx [hunk 17]
```diff
@@ -4,6 +4,30 @@ Note: the signIn result also includes an `AuthCodeDeliveryDetails` member. It in
...
+                if (result.isSignedIn()) {
+                    Log.i(TAG, "Confirm signIn succeeded");
+                } else {
+                    Log.i(TAG, "Confirm sign in not complete. There might be additional steps: "+ result.getNextStep());
+                    // Switch on the next step to take appropriate actions.
+                    // If `signInResult.isSignedIn` is true, the next step
+                    // is 'done', and the user is now signed in.
+                }
+            },
+             error -> Log.e(TAG, "Confirm sign in failed: " + error)
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx:23]:
```suggestion
            error -> Log.e(TAG, "Confirm sign in failed: " + error)
```

File: src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx [hunk 18]
```diff
@@ -4,6 +4,30 @@ Note: the signIn result also includes an `AuthCodeDeliveryDetails` member. It in
...
+            );
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/20_confirm_sms_mfa.mdx:24]:
```suggestion
    );
```

File: src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx [hunk 19]
```diff
@@ -2,6 +2,30 @@ If the next step is `CONFIRM_SIGN_IN_WITH_NEW_PASSWORD`, Amplify Auth requires a
...
+                    Log.i(TAG, "Confirm signIn succeeded");
+                } else {
+                    Log.i(TAG, "Confirm sign in not complete. There might be additional steps: " + result.getNextStep());
+                    // Switch on the next step to take appropriate actions.
+                    // If `signInResult.isSignedIn` is true, the next step
+                    // is 'done', and the user is now signed in.
+                }
+            },
+             error -> Log.e(TAG, "Confirm sign in failed: " + error)
+            );
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx:22]:
```suggestion
    );
```

File: src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx [hunk 20]
```diff
@@ -2,6 +2,30 @@ If the next step is `CONFIRM_SIGN_IN_WITH_CUSTOM_CHALLENGE`, Amplify Auth is awa
...
+    } catch (Exception error) {
+        Log.e(TAG, "Unexpected error: " + error);
+    }
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/30_confirm_custom_challenge.mdx:25]: This catch block is indented too far (it should line up with `try {`)

File: src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx [hunk 21]
```diff
@@ -2,6 +2,30 @@ If the next step is `CONFIRM_SIGN_IN_WITH_NEW_PASSWORD`, Amplify Auth requires a
...
+    } catch (Exception error) {
+        Log.e(TAG, "Unexpected error: " + error);
+    }
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/40_confirm_new_password.mdx:25]: This catch block is indented too far (it should line up with `try {`)

File: src/fragments/lib/auth/android/signin_next_steps/60_confirm_signup.mdx [hunk 22]
```diff
@@ -2,6 +2,22 @@ If you receive `CONFIRM_SIGN_UP` as a next step, sign up could not proceed witho
...
+ try {
+     Amplify.Auth.confirmSignUp(
+             "username",
+             "confirmation code",
+             result -> Log.i(TAG, "Confirm signUp result completed: " + result.isSignUpComplete()),
+             error -> Log.e(TAG, "An error occurred while confirming sign up: " + error)
+     );
+    } catch (Exception error) {
+       Log.e(TAG, "unexpected error: " + error);
+    }
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/60_confirm_signup.mdx:17]: This catch block is indented too far (it should line up with `try {`)

File: src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx [hunk 23]
```diff
@@ -4,9 +4,76 @@ The `nextStep` property is of enum type `AuthSignInStep`. Depending on its value
...
+                          // User has successfully signed in to the app
+                          break;
+                      }
+                  }
+              },
+              error -> Log.e(TAG, "SignIn failed: " + error)
+      );
+      } catch (Exception error) {
+          Log.e(TAG, "Unexpected error occurred: " + error);
+      }
```
Comment (eeatonaws) [inline src/fragments/lib/auth/android/signin_next_steps/10_signin.mdx:68]: This catch block is indented too far (it should line up with `try {`)

**Review by eeatonaws — CHANGES_REQUESTED** [review]
Left comments about additional formatting changes.

**Review by eeatonaws — APPROVED** [review]

**Pull request merged**